install-app.sh

```
@@ -5,16 +5,17 @@
 # runs scripts under root user. 
 
 # install system updates and isntall python3-pip package using apt. '-yq' flags are 
-# used to suppress any interactive prompts - we won't be able to confirm operation 
-# when running the script as VM extention.  
+# used to suppress any interactive prompts - we won't be able to confirm operation
```

The comment explains why -yq flags are used, which is fine. Add a defensive shell mode at the very top of the script (`set -euo pipefail`) so failures cause the extension to fail fast and return useful logs. Without that the extension may appear to succeed while commands silently fail.